x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -407,7 +407,9 @@ public static Object convert(Object value, DataType dataType) {
 
         NUMERIC_TO_BOOLEAN(fromLong(value -> value != 0)),
         STRING_TO_BOOLEAN(fromString(DataTypeConversion::convertToBoolean, "Boolean")),
-        DATE_TO_BOOLEAN(fromDate(value -> value != 0));
+        DATE_TO_BOOLEAN(fromDate(value -> value != 0)),
+
+        BOOL_TO_LONG(fromBool(value -> value ? 1L : 0L));
```

Comment:
My preference would be to move `BOOL_TO_LONG` up and start using it instead of `BOOL_TO_INT` for date conversions (we can't right now since the enum entry is defined after).
It's consistent and avoids the int to long conversion.